### PR TITLE
Check project packaging before execute `azure-spring:deploy` 

### PR DESCRIPTION
**Pull request by Flanker32**
Check project packaging before execute `azure-spring:deploy` 

`azure-spring:deploy` will skip pom project and throw exception for other projects which packing is not `pom` or `jar`

### CODE REVIEW

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 1]
```diff
@@ -25,8 +26,15 @@
 @Mojo(name = "deploy")
 public class DeployMojo extends AbstractSpringMojo {
 
+    protected static final String PROJECT_NOT_SUPPORT = "`azure-spring:deploy` does not support maven project with " +
+            "build type %s, only jar is supported";
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: build type => packaging

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 2]
```diff
@@ -46,6 +54,19 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
         getLog().info(deploymentClient.getDeploymentStatus().toString());
     }
 
+    protected boolean checkProjectPackaging(MavenProject project) throws MojoExecutionException {
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: protected static boolean isPomPackingProject(MavenProject proj) {
        return POM.equals(proj.getPackaging());
    }

     protected static boolean isJarPackingProject(MavenProject proj) {
        return JAR.equals(proj.getPackaging());
    }

Comment (andxu) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: add and use these two methods

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 1 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: Fixed

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 2 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: Done

**Pull request merged**